.circleci/config.yml

```diff
@@ -57,6 +55,18 @@ jobs:
           name: Publish Docker Image to Docker Hub
           command: |
             echo "${DOCKERHUB_PASS}" | docker login -u "${DOCKERHUB_USERNAME}" --password-stdin
+            TAG=${CIRCLE_TAG/v/''}
+            D_TAG="$(wget -q https://registry.hub.docker.com/v1/repositories/logdna/logdna-agent/tags -O -  | sed -e 's/[][]//g' -e 's/"//g' -e 's/ //g' | tr '}' '\n'  | awk -F: '{print $3}' | sort -g -r | head -1)"
+            lenght="$(echo $TAG | wc -m)"
+            if [ $lenght > 3 ]; then
+              IMAGE_TAG=$D_TAG
```

Comment:
I'd suggest to use brackets for variables too: `${D_TAG}` instead of `$D_TAG` - this makes thing much cleaner and safer.